data_adapter/preprocessing.py

```diff
@@ -265,7 +266,11 @@ def convert_series(series: list[float], factor: float) -> list[float]:
                 if "array" in field["type"]:
                     df[field["name"]] = df[field["name"]].apply(convert_series, factor=conversion_factor)
                 else:
-                    df[field["name"]] = df[field["name"]] * conversion_factor
+                    try:
+                        df[field["name"]] = df[field["name"]] * conversion_factor
+                    except TypeError:  # occurs for foreign keys
+                        # rewrite unit as conversion did not take place
+                        df_units[field["name"]] = field["unit"]
```

Comment:
I don't understand why this is needed? or why a TypeError is thrown.
In line 256 i see the same line, maybe check if conversion can be done should be moved upwards?

Reply:
Good point. I moved the check upwards, in line with the other checks.
TypeError is thrown because string (foreign key) cannot be multiplied by float.